File: backend/app/models/opportunity.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID, uuid4
from datetime import datetime
from enum import Enum


class OpportunityStatus(str, Enum):
    pending = "pending"
    approved = "approved"
    rejected = "rejected"
    sent = "sent"


class Channel(str, Enum):
    hackernews = "hackernews"
    reddit = "reddit"
    linkedin = "linkedin"
    twitter = "twitter"


class Opportunity(BaseModel):
    id: UUID = Field(default_factory=uuid4)
    product_id: UUID
    user_id: UUID
    channel: Channel
    source_url: str
    source_title: str
    source_body: str
    relevance_score: float = Field(ge=0.0, le=1.0)
    roi_score: float = Field(ge=0.0, le=1.0)
    draft: Optional[str] = None
    status: OpportunityStatus = OpportunityStatus.pending
    created_at: datetime = Field(default_factory=datetime.utcnow)
    acted_at: Optional[datetime] = None


class OpportunityCreate(BaseModel):
    product_id: UUID
    user_id: UUID
    channel: Channel
    source_url: str
    source_title: str
    source_body: str

File: android/app/src/main/java/com/lowlevellogic/cast/data/models/Opportunity.kt
package com.lowlevellogic.cast.data.models

import com.google.gson.annotations.SerializedName

data class Opportunity(
    val id: String,
    @SerializedName("product_id") val productId: String,
    val channel: String,
    @SerializedName("source_url") val sourceUrl: String,
    @SerializedName("source_title") val sourceTitle: String,
    @SerializedName("source_body") val sourceBody: String,
    @SerializedName("relevance_score") val relevanceScore: Float,
    @SerializedName("roi_score") val roiScore: Float,
    val draft: String?,
    val status: String,
    @SerializedName("created_at") val createdAt: String,
)
